TicTacToe.py

```diff
@@ -216,23 +216,46 @@ def resetGame(self):
 		self.board = self.emptyBoard()
 		self.game_state = self.NO_WINNER
 
+	def updatePlayerIcons(self, player_0_icon, player_1_icon):
+		self.PLAYER_0_ICON = player_0_icon
+		self.PLAYER_1_ICON = player_1_icon
+
 	# behind the scenes functions
 	##########################################################################################
 	# terminal functions
 
-	def terminalGame(self):
-		"""Prints messages to allow the user to select number of players and player icon,
-		then begins and runs the game by calling the necessary functions.
+	def advancedGameSettings(self, settingToChange):
+		match settingToChange:
+			case 'change icons':
+				print('Both player icons must only be 1 character long.')
+				player0 = 'too long'
+				while len(player0) != 1:
+					player0 = input('What do you want first move icon to be? (Traditionaly X): ')
```

Comment:
If user input is greater than one character long, should there be a message to say "Player icons must be one character long?"  (I know you don't like that redundancy because "read directions the first time" but count on me to always want it 😄)

```suggestion
					player0 = input('What do you want first move icon to be? (Traditionaly X): ')
					if len(player0) != 1:
						print("Please enter a single character for the player icon")
```